.github/workflows/_reusable_run_tests.yml

```diff
@@ -220,15 +226,21 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
   upload-test-results-datadadog:
+    name: Upload to Datadog
     runs-on: ubuntu-latest
     needs: run-tests
     if: always()
     steps:
+      - name: Sanitize branch name
+        run: |
+            SANITIZED_BRANCH=$(echo "${{ inputs.branch }}" | sed 's/[^a-zA-Z0-9._-]/_/g')
+            echo "SANITIZED_BRANCH=$SANITIZED_BRANCH" >> $GITHUB_ENV
+
       - name: Download tests results
-        uses: actions/download-artifact@v3
+        uses: actions/download-artifact@v4
         continue-on-error: true
         with:
-          name: test-reports
+          name: Test reports for ${{ env.SANITIZED_BRANCH }} (${{ github.run_id }})
```

Comment:
optional: Consider extracting to a local [composite action](https://docs.github.com/en/actions/sharing-automations/creating-actions/creating-a-composite-action)

Reply:
yeah, I started to look into that, I'll rework it a bit in another PRs